velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -209,4 +209,31 @@ struct LastDayFunction {
   }
 };
 
+template <typename T>
+struct DateSubFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
```

Comment:
Consider making this a template to avoid repeating the function.

```
template <typename T>
FOLLY_ALWAYS_INLINE void call(
      out_type<Date>& result,
      const arg_type<Date>& date,
      T value) {
    int32_t subValue = 0 - value;
    result = addToDate(date, DateTimeUnit::kDay, subValue);
  }
```

Reply:
Done